packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts

```diff
@@ -0,0 +1,227 @@
+        ? Resource.empty()
+        : new Resource({
+            [K8S_RESOURCE.CLUSTER_NAME]: clusterName || '',
+            [CONTAINER_RESOURCE.ID]: containerId || '',
+          });
+    } catch (e) {
+      config.logger.warn('Process is not running on K8S', e);
+      return Resource.empty();
+    }
+  }
+
+  /**
+   * Attempts to make a connection to AWS Config map which will
+   * determine whether the process is running on an Eks
+   * process if the config map is empty or not
+   * @param config The resource detection config with a required logger
+   */
+  private async _isEks(
+    config: ResourceDetectionConfigWithLogger,
+    cert: Buffer
+  ): Promise<boolean> {
+    const options = {
+      hostname: this.K8S_SVC_URL,
+      path: this.AUTH_CONFIGMAP_PATH,
+      method: 'GET',
+      timeout: this.MILLISECOND_TIME_OUT,
+      headers: {
+        Authorization: await this._getK8sCredHeader(config),
+      },
+      ca: cert,
+    };
+    return !!(await this._fetchString(options));
+  }
+
+  /**
+   * Attempts to make a connection to Amazon Cloudwatch
+   * Config Maps to grab cluster name
+   * @param config The resource detection config with a required logger
+   */
+  private async _getClusterName(
+    config: ResourceDetectionConfigWithLogger,
+    cert: Buffer
+  ): Promise<string | undefined> {
+    const options = {
+      host: this.K8S_SVC_URL,
+      path: this.CW_CONFIGMAP_PATH,
+      method: 'GET',
+      timeout: this.MILLISECOND_TIME_OUT,
+      headers: {
+        Authorization: await this._getK8sCredHeader(config),
+      },
+      ca: cert,
+    };
+    const response = await this._fetchString(options);
+    try {
+      return JSON.parse(response).data['cluster.name'];
+    } catch (e) {
+      config.logger.warn('Cannot get cluster name on EKS', e);
+    }
+    return '';
+  }
+  /**
+   * Reads the Kubernetes token path and returns kubernetes
+   * credential header
+   * @param config The resource detection config with a required logger
+   */
+  private async _getK8sCredHeader(
+    config: ResourceDetectionConfigWithLogger
+  ): Promise<string> {
+    try {
+      const content = await AwsEksDetector.readFileAsync(
+        this.K8S_TOKEN_PATH,
+        'utf8'
+      );
+      return 'Bearer ' + content;
+    } catch (e) {
+      config.logger.warn('Unable to read Kubernetes client token.', e);
+    }
+    return '';
+  }
+
+  /**
+   * Read container ID from cgroup file
+   * In EKS, even if we fail to find target file
+   * or target file does not contain container ID
+   * we do not throw an error but throw warning message
+   * and then return null string
+   */
+  private async _getContainerId(
+    config: ResourceDetectionConfigWithLogger
+  ): Promise<string | undefined> {
+    try {
+      const rawData = await AwsEksDetector.readFileAsync(
+        this.DEFAULT_CGROUP_PATH,
+        'utf8'
+      );
+      const splitData = rawData.trim().split('\n');
+      for (const str of splitData) {
```

Comment:
can you explain why the length has to be greater, how will the `rawData` looks like - does it have some predefined structure, maybe a link to documentation where this is described would be helpful or if the documentation is too big please provide some description in ts doc for better understanding.

Reply:
In the JS [ECS Resource detector](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts#L64) and Java [EKS Resource detector](https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/resource/DockerHelper.java#L38) the same way to read the container ID was implemented I thought it would be proper to keep it consistent.

Reply:
Ah, the docker-generated cgroup file returns a structure that lists the untruncated docker Container ID at the end of each line. The predefined structure of calling /proc/self/cgroup when in a docker container has the structure:

#:xxxxxx:/
or 
#:xxxxxx:/docker/64characterID

For example on a docker process running cat cgroup/self/proc returns a predefined structure:

3:memory:/
2:cpu,cpuacct:/docker/79311de543c2b01bdbb7ccaf355e71a02b0726366a1427ecfceb5e1f5be81644
1:name=systemd:/docker/79311de543c2b01bdbb7ccaf355e71a02b0726366a1427ecfceb5e1f5be81644

So this function takes advantage of the fact that reading the last 64 characters of an untruncated line returns the Docker ID.

For Reference: checking if a process is running on [docker
](https://stackoverflow.com/questions/23513045/how-to-check-if-a-process-is-running-inside-docker-container)or parsing the [id](https://forums.docker.com/t/get-a-containers-full-id-from-inside-of-itself/37237) from cgroup.

Reply:
Just please add this info to the doc of the function and we are good to go :)

Reply:
Cheers :)